kratos/factories/base_factory.h

```diff
@@ -0,0 +1,270 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Vicente Mataix Ferrandiz
+//
+
+#if !defined(KRATOS_BASE_FACTORY_H_INCLUDED )
+#define  KRATOS_BASE_FACTORY_H_INCLUDED
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/define.h"
+#include "includes/kratos_parameters.h"
+#include "includes/kratos_components.h"
+#include "includes/shared_pointers.h"
+#include "solving_strategies/strategies/explicit_solving_strategy.h"
+#include "solving_strategies/builder_and_solvers/explicit_builder.h"
+#include "spaces/ublas_space.h"
+
+namespace Kratos
+{
+///@name Type Definitions
+///@{
+
+typedef TUblasSparseSpace<double> SparseSpaceType;
+typedef TUblasDenseSpace<double> LocalSpaceType;
+
+///@}
+///@name Kratos Classes
+///@{
+
+/**
+ * @class BaseFactoryMethods
+ * @ingroup KratosCore
+ * @brief Here we define some common methods
+ * @details Defines the base class factory methods
+ * @author Vicente Mataix Ferrandiz
+ */
+class BaseFactoryMethods
```

Comment:
why is this class necessary?

Reply:
In order to have some common methods, yes

Reply:
can you explain the class structure of the Factories please? Ideally with a UML diagramm?